server/methods/accounts/accounts.app-test.js

```diff
@@ -56,100 +68,90 @@ describe("Account Meteor method ", function () {
   function spyOnMethod(method, id) {
     return sandbox.stub(Meteor.server.method_handlers, `cart/${method}`, function (...args) {
       check(args, [Match.Any]); // to prevent audit_arguments from complaining
-      this.userId = id;
+      this.userId = id; // having to do this makes me think that we should be using Meteor.userId() instead of this.userId in our Meteor methods
       return originals[method].apply(this, args);
     });
   }
 
   describe("addressBookAdd", function () {
     beforeEach(function () {
-      Cart.remove({});
-      Accounts.remove({});
+      const sessionId = Random.id(); // Required for creating a cart
+      // editing your address book also udpates your cart, so be sure there
+      // is a cart present
+      Meteor.call("cart/createCart", fakeAccount.userId, sessionId);
     });
 
-    it("should allow user to add new addresses", function (done) {
-      let account = Factory.create("account");
-      sandbox.stub(Meteor, "userId", function () {
-        return account.userId;
-      });
+    it("should allow user to add new addresses", function () {
       const address = getAddress();
       // we already have one address by default
-      expect(account.profile.addressBook.length).to.equal(1);
+      expect(fakeAccount.profile.addressBook.length).to.equal(1);
+
       Meteor.call("accounts/addressBookAdd", address);
-      account = Accounts.findOne(account._id);
+
+      const account = Accounts.findOne(fakeAccount._id);
```

Comment:
Nitpick, but we are trying to move away from the shorthand `findOne(id)` to the more explicit `findOne({ _id: fakeAccount._id });`